### ISSUE TITLE
ARM:topology_test:test_downgrade_after_partial_upgrade debug mode

### ISSUE THREAD

**Issue by Annamikhlin** [scylladb/scylladb#15387]
`test_cluster_features.test_downgrade_after_partial_upgrade` failed on ARM debug mode with error:

```
=================================== FAILURES ===================================
_____________________ test_downgrade_after_partial_upgrade _____________________

manager = <test.pylib.manager_client.ManagerClient object at 0xffff33f86510>

    @pytest.mark.asyncio
    async def test_downgrade_after_partial_upgrade(manager: ManagerClient) -> None:
        """Simulates a partial upgrade of a cluster by enabling the test features
           in all nodes but one, then downgrading the upgraded nodes.
        """
        servers = await manager.running_servers()
        upgrading_servers = servers[1:] if len(servers) > 0 else []
    
        cql = manager.cql
    
        # Upgrade
        for srv in upgrading_servers:
            await change_support_for_test_feature_and_restart(manager, [srv], enable=True)
            host = (await wait_for_cql_and_get_hosts(cql, [srv], time.time() + 60))[0]
            assert TEST_FEATURE_NAME in await get_supported_features(cql, host)
    
        # There is one node that is not upgraded. The feature should not be enabled.
        for srv in servers:
            assert TEST_FEATURE_NAME not in await get_enabled_features(cql, host)
    
        # Downgrade, in reverse order
        for srv in upgrading_servers[::-1]:
>           await change_support_for_test_feature_and_restart(manager, [srv], enable=False)

test/topology/test_cluster_features.py:117: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/topology/test_cluster_features.py:59: in change_support_for_test_feature_and_restart
    await asyncio.gather(*(manager.server_start(srv.server_id, expected_error) for srv in srvs))
test/pylib/manager_client.py:155: in server_start
    await self.client.get_text(f"/cluster/server/{server_id}/start", params=params)
test/pylib/rest_client.py:82: in get_text
    ret = await self._fetch("GET", resource_uri, response_type = "text", host = host,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test.pylib.rest_client.UnixRESTClient object at 0xffff351b21d0>
method = 'GET', resource = '/cluster/server/388/start', response_type = 'text'
host = None, port = None, params = None, json = None, timeout = None

    async def _fetch(self, method: str, resource: str, response_type: Optional[str] = None,
                     host: Optional[str] = None, port: Optional[int] = None,
                     params: Optional[Mapping[str, str]] = None,
                     json: Optional[Mapping] = None, timeout: Optional[float] = None) -> Any:
        # Can raise exception. See https://docs.aiohttp.org/en/latest/web_exceptions.html
        assert method in ["GET", "POST", "PUT", "DELETE"], f"Invalid HTTP request method {method}"
        assert response_type is None or response_type in ["text", "json"], \
                f"Invalid response type requested {response_type} (expected 'text' or 'json')"
        # Build the URI
        port = port if port else self.default_port if hasattr(self, "default_port") else None
        port_str = f":{port}" if port else ""
        assert host is not None or hasattr(self, "default_host"), "_fetch: missing host for " \
                "{method} {resource}"
        host_str = host if host is not None else self.default_host
        uri = self.uri_scheme + "://" + host_str + port_str + resource
        logging.debug(f"RESTClient fetching {method} {uri}")
    
        client_timeout = ClientTimeout(total = timeout if timeout is not None else 300)
        async with request(method, uri,
                           connector = self.connector if hasattr(self, "connector") else None,
                           params = params, json = json, timeout = client_timeout) as resp:
            if resp.status != 200:
                text = await resp.text()
>               raise HTTPError(uri, resp.status, params, json, text)
E               test.pylib.rest_client.HTTPError: HTTP error 500, uri: http+unix://api/cluster/server/388/start, params: None, json: None, body:
E               failed to start the node, server_id 388, IP 127.115.81.62, workdir scylla-388, host_id f0f4cf91-f3a4-4ea3-9ed5-53e21875c095, cql [not connected]
E               Check the log files:
E               /scylladir/testlog/aarch64/test.py.debug-release-dev.log
E               /scylladir/testlog/aarch64/debug/scylla-388.log

test/pylib/rest_client.py:69: HTTPError
```

link to the failure: https://jenkins.scylladb.com/job/scylla-master/job/build/1623/artifact/testlog/aarch64/debug/topology.test_cluster_features.1.log/

**Comment by kbr-scylla** [comment]
Please upload the logs directory to github so we don't lose them once Jenkins clears the job history.

**Comment by Annamikhlin** [comment]
added: [topology.test_cluster_features.1.log](https://github.com/scylladb/scylladb/files/12646019/topology.test_cluster_features.1.log)

**Comment by benipeled** [comment]
Again https://jenkins.scylladb.com/job/scylla-enterprise/job/build/1064/testReport/junit/(root)/test_cluster_features/Build__Test__Publish___buildTestPublishArm___test_downgrade_after_partial_upgrade/

**Comment by mykaul** [comment]
> Again https://jenkins.scylladb.com/job/scylla-enterprise/job/build/1064/testReport/junit/(root)/test_cluster_features/Build__Test__Publish___buildTestPublishArm___test_downgrade_after_partial_upgrade/

```
INFO  2023-10-31 01:26:02,334 [shard 0:main] seastar - updated: blocked-reactor-notify-ms=25
INFO  2023-10-31 01:26:02,335 [shard 1:main] seastar - updated: blocked-reactor-notify-ms=25
INFO  2023-10-31 01:26:02,336 [shard 0:main] init - starting storage proxy
Reactor stalled for 36 ms on shard 0. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0x37c18c3 0x37c186f 0x37c17bb 0x37c155f 0x390386f 0x38e7983 0x17bcf74b 0xdcf65a3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3cb4fcb 0x3715c5b 0x3713387 0xdbe39b7 0xeca30a7 0xeca2c6b 0xeca247b 0xde09c3f 0xdbe0be3 0x30597 0x3066f 0xdafb7af
Reactor stalled for 35 ms on shard 0. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0xa9683 0xdafe46b 0xdafc44f 0xdbd65df 0x3933f33 0x393366b 0x3931ee3 0x390b16f 0x38f2537 0x38f559b 0x38e71eb 0x17bb52f3 0x17bc184b 0xdcf65f3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3cb4fcb 0x3715c5b 0x3713387 0xdbe39b7 0xeca30a7 0xeca2c6b 0xeca247b 0xde09c3f 0xdbe0be3 0x30597 0x3066f 0xdafb7af
Reactor stalled for 34 ms on shard 0. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0x343ad03 0x343a203 0x3439a97 0x3438f83 0x3435223 0x3932f43 0x3932db7 0x390af13 0x38f2537 0x38f559b 0x38e71eb 0x17bcede7 0xdcf65a3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3cb4fcb 0x3715c5b 0x3713387 0xdbe39b7 0xeca30a7 0xeca2c6b 0xeca247b 0xde09c3f 0xdbe0be3 0x30597 0x3066f 0xdafb7af
Reactor stalled for 35 ms on shard 0. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0xdb080b7 0x3925673 0x3924aab 0x39395a7 0x3938e0b 0x39382ab 0x3937da7 0x39377ff 0x394eadf 0x39bc2ef 0x39b9207 0x39b706f 0x39b6667 0x391a7d3 0x38f3abf 0x38f24bb 0x38f559b 0x38e71eb 0x17bcede7 0xdcf65a3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3cb4fcb 0x3715c5b 0x3713387 0xdbe39b7 0xeca30a7 0xeca2c6b 0xeca247b 0xde09c3f 0xdbe0be3 0x30597 0x3066f 0xdafb7af
Reactor stalled for 35 ms on shard 0. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0xde285c7 0x35fb77f 0x39257f3 0x393b75b 0x39e80d3 0x390c9bb 0x38f2fb7 0x38f24bb 0x38f559b 0x38e71eb 0x17bb52f3 0x17bb4a5b 0x17bc1743 0xdcf65f3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3cb4fcb 0x3715c5b 0x3713387 0xdbe39b7 0xeca30a7 0xeca2c6b 0xeca247b 0xde09c3f 0xdbe0be3 0x30597 0x3066f 0xdafb7af
Reactor stalled for 34 ms on shard 1. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0xdbc6433 0xdafe273 0xdafbfa7 0xdb9c68f 0x3427a87 0x343421b 0x3428f43 0x3904087 0x38e82e7 0x17bcea23 0xdcf65a3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3ded70f 0x3debe67 0x3debb83 0x3deae13 0xeee16db 0x3a0590b 0x8bc53 0xf929b
Reactor stalled for 34 ms on shard 1. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0xdbc6b5b 0xdba76f3 0xdbd6e4b 0xdeb9d1b 0xdeb9cbf 0xdeb7c4b 0xdeafcc7 0x17bde0df 0x17bb5203 0x17bb4a07 0x17bc1743 0xdcf65f3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3ded70f 0x3debe67 0x3debb83 0x3deae13 0xeee16db 0x3a0590b 0x8bc53 0xf929b
Reactor stalled for 36 ms on shard 1. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0xdbc63e3 0xdafe273 0xdafc44f 0xdbd65df 0x343c287 0x343bf23 0x343bdcb 0x392a16f 0x3929a0f 0x3928cc7 0x3928fc7 0x39286c3 0x3927cbb 0x3905c3b 0x3905a3f 0x38ed8d7 0x39ee673 0x39ee1d3 0x3918ab7 0x38f2c0f 0x38f24bb 0x38f559b 0x38e71eb 0x17bb52f3 0x17bb4aaf 0x17bc1743 0xdcf65f3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3ded70f 0x3debe67 0x3debb83 0x3deae13 0xeee16db 0x3a0590b 0x8bc53 0xf929b
Reactor stalled for 32 ms on shard 1. Backtrace: 0xdb417e3 0x3ea436f 0x3ea3777 0x3c89267 0x3c8351b 0x3c82f77 0x3c8394b 0x3c8a883 0x7db 0x341cf1f 0x3427837 0x343c493 0x343c343 0x343c113 0x343be2f 0x392a16f 0x3929a0f 0x3928cc7 0x39286c3 0x3927cbb 0x3905c3b 0x3905a3f 0x38ed8d7 0x39ee673 0x39ee1d3 0x3918ab7 0x38f2c0f 0x38f24bb 0x38f559b 0x38e71eb 0x17bcadc7 0xdcf65a3 0xdcf644b 0xdcf60df 0xdcf56f3 0xe4288f3 0x3ce52a3 0x3df1a9f 0x3df312b 0x3df2f0f 0x3df2cfb 0x3df2ad3 0x3df28a7 0x3df27ff 0x3df2673 0x3df23d3 0x3caa5cf 0x3cb32e3 0x3cb6ecf 0x3ded70f 0x3debe67 0x3debb83 0x3deae13 0xeee16db 0x3a0590b 0x8bc53 0xf929b
AddressSanitizer:DEADLYSIGNAL
=================================================================
==24375==ERROR: AddressSanitizer: SEGV on unknown address 0xfffffffffffffff8 (pc 0xffff89ecd174 bp 0xffff846c9e30 sp 0xffff846c9e30 T1)
==24375==The signal is caused by a READ memory access.
AddressSanitizer:DEADLYSIGNAL
AddressSanitizer: nested bug in the same thread, aborting.
------ Ending test debug/topology.test_cluster_features.1::test_downgrade_after_partial_upgrade ------
```

Thought we were over this... @avikivity  ?

**Comment by mykaul** [comment]
(relevant log - https://jenkins.scylladb.com/job/scylla-enterprise/job/build/1064/artifact/testlog/aarch64/debug/scylla-142.log )

**Comment by mykaul** [comment]
Reactor stalls:
```
Reactor stalled for 36 ms on shard 0. Backtrace:
[Backtrace #0]
__interceptor_backtrace at ??:?
$d.34 at server.cc:?
$d.34 at server.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
?? ??:0
$d.5481 at writer.cc:?
$d.5481 at writer.cc:?
$d.5481 at writer.cc:?
$d.5481 at writer.cc:?
$d.10486 at reader.cc:?
$d.10486 at reader.cc:?
service::storage_proxy_stats::stats::register_stats() at ./service/storage_proxy.cc:2733
operator() at ./main.cc:1098
void std::__invoke_impl<void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&, {lambda()#2}>(std::__invoke_other, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&, {lambda()#2}&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:61
std::enable_if<is_invocable_r_v<void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&, {lambda()#2}>, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&>::type std::__invoke_r<void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&, {lambda()#2}>(std::enable_if&&, (void&&)...) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:111
std::_Function_handler<void (void*), scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}>::_M_invoke(std::_Any_data const&, void*&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:290
std::function<void (void*)>::operator()(void*) const at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
$d.567 at prepended_input_stream.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1029 at incremental_backlog_tracker.cc:?
$d.1029 at incremental_backlog_tracker.cc:?
$d.1029 at incremental_backlog_tracker.cc:?
$d.1029 at incremental_backlog_tracker.cc:?
$d.10501 at reader.cc:?
$d.10501 at reader.cc:?
scylla_main(int, char**) at ./main.cc:685
int std::__invoke_impl<int, int (*&)(int, char**), int, char**>(std::__invoke_other, int (*&)(int, char**), int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:61
std::enable_if<is_invocable_r_v<int, int (*&)(int, char**), int, char**>, int>::type std::__invoke_r<int, int (*&)(int, char**), int, char**>(int (*&)(int, char**), int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:114
std::_Function_handler<int (int, char**), int (*)(int, char**)>::_M_invoke(std::_Any_data const&, int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:290
std::function<int (int, char**)>::operator()(int, char**) const at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
main at ./main.cc:2053
?? ??:0
?? ??:0
_start at ??:?
```

I wonder if we should wait with:
```
INFO  2023-10-31 01:26:02,334 [shard 0:main] seastar - updated: blocked-reactor-notify-ms=25
INFO  2023-10-31 01:26:02,335 [shard 1:main] seastar - updated: blocked-reactor-notify-ms=25
```

Or those are unrelated?

**Comment by mykaul** [comment]
(unsure why the backtrace is so bad, it's a debug build, what's missing?)

**Comment by mykaul** [comment]
Another not-so-decoded backtrace:
```
<div style="color: rgb(0, 0, 0); font-family: Arial, Helvetica, sans-serif; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; padding: 10px;"><form id="index" action="http://backtrace.scylladb.com/index.html" method="post" enctype="application/x-www-form-urlencoded">

Backtrace: |  
-- | --
  |  

</form></div><div style="color: rgb(0, 0, 0); font-family: Arial, Helvetica, sans-serif; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; padding: 10px;"><pre>Reactor stalled for 32 ms on shard 1. Backtrace:
[Backtrace #0]
__interceptor_backtrace at ??:?
$d.34 at server.cc:?
$d.34 at server.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
$d.2783 at incremental_compaction_strategy.cc:?
?? ??:0
$d.148 at sstables.cc:?
$d.148 at sstables.cc:?
$d.148 at sstables.cc:?
$d.148 at sstables.cc:?
$d.148 at sstables.cc:?
$d.148 at sstables.cc:?
$d.1405 at compress.cc:?
$d.1405 at compress.cc:?
$d.1405 at compress.cc:?
$d.1405 at compress.cc:?
$d.1405 at compress.cc:?
$d.10486 at reader.cc:?
$d.10486 at reader.cc:?
$d.10486 at reader.cc:?
$d.45 at compaction.cc:?
$d.45 at compaction.cc:?
$d.656 at sstable_version.cc:?
$d.10486 at reader.cc:?
$d.10486 at reader.cc:?
$d.10486 at reader.cc:?
$d.10486 at reader.cc:?
service::storage_proxy_stats::stats::register_stats() at ./service/storage_proxy.cc:2557
operator() at ./main.cc:1098
void std::__invoke_impl&lt;void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&amp;, {lambda()#2}&gt;(std::__invoke_other, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&amp;, {lambda()#2}&amp;&amp;) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:61
std::enable_if&lt;is_invocable_r_v&lt;void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&amp;, {lambda()#2}&gt;, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&amp;&gt;::type std::__invoke_r&lt;void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&amp;, {lambda()#2}&gt;(std::enable_if&amp;&amp;, (void&amp;&amp;)...) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:111
std::_Function_handler&lt;void (void*), scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&gt;::_M_invoke(std::_Any_data const&amp;, void*&amp;&amp;) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:290
std::function&lt;void (void*)&gt;::operator()(void*) const at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
$d.567 at prepended_input_stream.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1029 at incremental_backlog_tracker.cc:?
$d.1029 at incremental_backlog_tracker.cc:?
$d.1029 at incremental_backlog_tracker.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
$d.1773 at metadata_collector.cc:?
std::function&lt;void ()&gt;::operator()() const at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
$d.45 at compaction.cc:?
?? ??:0
?? ??:0</pre></div>
```

**Comment by avikivity** [comment]
Reactor stall isn't a problem.

@gusev-p didn't we disable the stall detector in debug mode?

**Comment by avikivity** [comment]
Ah, blocked on https://github.com/scylladb/scylla-enterprise/issues/3536

**Comment by mykaul** [comment]
> Reactor stall isn't a problem.

I was more concerned with the ASAN 0xfffffffffffffff8 issue, or is it related to the stall detector?

**Comment by gusev-p** [comment]
> Reactor stall isn't a problem.
> 
> @gusev-p didn't we disable the stall detector in debug mode?

> INFO  2023-10-31 01:20:55,871 [shard 0:main] seastar - updated: blocked-reactor-notify-ms=1000000

There is no my commit in this build, 1000000 millis was before it.

**Comment by gusev-p** [comment]
> Reactor stalls:
> 
> ```
> Reactor stalled for 36 ms on shard 0. Backtrace:
> [Backtrace #0]
> __interceptor_backtrace at ??:?
> $d.34 at server.cc:?
> $d.34 at server.cc:?
> $d.2783 at incremental_compaction_strategy.cc:?
> $d.2783 at incremental_compaction_strategy.cc:?
> $d.2783 at incremental_compaction_strategy.cc:?
> $d.2783 at incremental_compaction_strategy.cc:?
> $d.2783 at incremental_compaction_strategy.cc:?
> ?? ??:0
> $d.5481 at writer.cc:?
> $d.5481 at writer.cc:?
> $d.5481 at writer.cc:?
> $d.5481 at writer.cc:?
> $d.10486 at reader.cc:?
> $d.10486 at reader.cc:?
> service::storage_proxy_stats::stats::register_stats() at ./service/storage_proxy.cc:2733
> operator() at ./main.cc:1098
> void std::__invoke_impl<void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&, {lambda()#2}>(std::__invoke_other, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&, {lambda()#2}&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:61
> std::enable_if<is_invocable_r_v<void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&, {lambda()#2}>, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&>::type std::__invoke_r<void, scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}&, {lambda()#2}>(std::enable_if&&, (void&&)...) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:111
> std::_Function_handler<void (void*), scylla_main(int, char**)::$_0::operator()() const::{lambda()#2}::operator()() const::{lambda(void*)#1}>::_M_invoke(std::_Any_data const&, void*&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:290
> std::function<void (void*)>::operator()(void*) const at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
> $d.567 at prepended_input_stream.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1773 at metadata_collector.cc:?
> $d.1029 at incremental_backlog_tracker.cc:?
> $d.1029 at incremental_backlog_tracker.cc:?
> $d.1029 at incremental_backlog_tracker.cc:?
> $d.1029 at incremental_backlog_tracker.cc:?
> $d.10501 at reader.cc:?
> $d.10501 at reader.cc:?
> scylla_main(int, char**) at ./main.cc:685
> int std::__invoke_impl<int, int (*&)(int, char**), int, char**>(std::__invoke_other, int (*&)(int, char**), int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:61
> std::enable_if<is_invocable_r_v<int, int (*&)(int, char**), int, char**>, int>::type std::__invoke_r<int, int (*&)(int, char**), int, char**>(int (*&)(int, char**), int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:114
> std::_Function_handler<int (int, char**), int (*)(int, char**)>::_M_invoke(std::_Any_data const&, int&&, char**&&) at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:290
> std::function<int (int, char**)>::operator()(int, char**) const at /usr/bin/../lib/gcc/aarch64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
> main at ./main.cc:2053
> ?? ??:0
> ?? ??:0
> _start at ??:?
> ```
> 
> I wonder if we should wait with:
> 
> ```
> INFO  2023-10-31 01:26:02,334 [shard 0:main] seastar - updated: blocked-reactor-notify-ms=25
> INFO  2023-10-31 01:26:02,335 [shard 1:main] seastar - updated: blocked-reactor-notify-ms=25
> ```
> 
> Or those are unrelated?

This is the same problem I investigated [here](https://github.com/scylladb/scylladb/issues/15090). The problem was that stall detector uses `glibc` `backtrace` function, which is not async signal safe. We [disabled stall detector](https://github.com/scylladb/scylladb/pull/15720) in debug mode as a mitigation, but the commit is not merged into enterprise yet, I guess.

**Comment by kbr-scylla** [comment]
> but the commit is not merged into enterprise yet, I guess.

Indeed.

The last master->enterprise merge is scylladb/scylla-enterprise@8f9683b62dd26146d3cea166d303df3235bd64d5.
It does not contain 9b1dfad51c4a60de44d41722aaa204a6f855a83a.

We'll keep hitting this on enterprise branch until a master->enterprise merge happens.

**Comment by mykaul** [comment]
> > but the commit is not merged into enterprise yet, I guess.
> 
> Indeed.
> 
> The last master->enterprise merge is [scylladb/scylla-enterprise@8f9683b](https://github.com/scylladb/scylla-enterprise/commit/8f9683b62dd26146d3cea166d303df3235bd64d5). It does not contain [9b1dfad](https://github.com/scylladb/scylladb/commit/9b1dfad51c4a60de44d41722aaa204a6f855a83a).
> 
> We'll keep hitting this on enterprise branch until a master->enterprise merge happens.

@avikivity ?

**Comment by mykaul** [comment]
I assume this is done long ago - and I haven't seen complaints on it in the last month or so.
